spring-kafka/src/main/java/org/springframework/kafka/support/converter/MessagingMessageConverter.java

```diff
@@ -66,13 +71,23 @@ public void setGenerateTimestamp(boolean generateTimestamp) {
 		this.generateTimestamp = generateTimestamp;
 	}
 
+	/**
+	 * Set the header mapper to map headers.
+	 * @param headerMapper the mapper.
+	 * @since 2.1
```

Comment:
Yes - 1.3 with headers, admin, and new threading from 2.0 (remove listener thread) - that was an early commit and might not be too hard to backport - it's the ConsumerAware stuff that's hard to backport.